main.cpp

```diff
@@ -672,6 +672,7 @@ struct CLIArguments
 	bool msl_emulate_subgroups = false;
 	uint32_t msl_fixed_subgroup_size = 0;
 	bool msl_force_sample_rate_shading = false;
+	bool msl_check_discarded_frag_stores = true;
```

Comment:
Should this be true? Realistically I think only MoltenVK will care about this. This is a bug that only appears for very specific CTS use cases, right?

Reply:
Maybe we need to split the workarounds into two, there's two problems that need to be solved:

- simd_is_helper_thread() being broken. The workaround for that is not invasive. I think we can apply it by default until a fix is known.
- Masking stores is exceptionally invasive and ugly.

Reply:
Done.